### PR TITLE
Improve explanations of usernames

### PR DESCRIPTION
**Pull request by kjetilk**
This should implement #823 and also address (at least partially) the bug in https://github.com/solid/solid/issues/205 by downcasing usernames on submition.

### CODE REVIEW

File: default-views/account/register-form.hbs [hunk 1]
```diff
@@ -9,7 +9,30 @@
...
                    required/>
-          </div>
+
+	    {{#if multiuser}}
+	    <p>Your username should be a lower-case word with only
+	    letters a-z and numbers 0-9 and without periods.</p>
+	    <p>Your public Solid POD URL will be:
+	    <tt>https://<span class="editable-username">alice</span>.<script type="text/javascript">
+              document.write(window.location.host)
+	    </script></tt></p>
```
Comment (RubenVerborgh) [inline default-views/account/register-form.hbs]: We could probably do this with server-side templating (passing in the hostname from `renderForm` in `create-account-request.js` if not already passed in). But this works too.

Comment (kjetilk) [reply default-views/account/register-form.hbs]: Yeah, I looked into that, but it seems we'd have to hook in the `SolidHost` class to do it, and I figured it is not worth it.

**Comment by kjetilk** [comment]
Right. It is just that stuff seems to happen every time I rebase, and I end up having to do a merge anyway...

**Comment by RubenVerborgh** [comment]
@kjetilk Ping me if I can help.

**Comment by kjetilk** [comment]
Yup, I will @RubenVerborgh , just need to investigate a bit more myself, to see if I understand the pain point.

BTW, the Snyk failure, is there something we could cherry-pick for that?

**Comment by RubenVerborgh** [comment]
@kjetilk Just a rebase on develop will do. The following should work:
```
git fetch
git checkout fix-username-docs
git rebase -i origin/develop
```

Then pick only the commits you need.

**Comment by kjetilk** [comment]
OK, so perhaps this is what happened: I merged the 5.0.0 branch into it...? If so, that was certainly an accident. I probably meant to merge develop into the 5.0.0 at that point.

Anyway, let me see if I can rebase it successfully with that understanding.

**Comment by kjetilk** [comment]
OK, so how about now?

**Comment by kjetilk** [comment]
OK, I usually look at commits as the gory details, but let me do that to experience the difference :-)

**Pull request merged**